tests/repository/test_upload.py

```diff
@@ -0,0 +1,118 @@
+import logging
```

Comment:
Do we have the FakeController/Client update yet?  I'm not seeing a related test.

Reply:
Forget that. Doing it now.